scripts/mpi_support/install_openmpi_host_injection.sh

```diff
@@ -0,0 +1,293 @@
+                ;;
+            --mpi-path)
+                if [ -n "$2" ]; then
+                    readonly MPI_PATH="$2"
+                    shift 2
+                else
+                    echo_red "Error: Argument required for $1"
+                    show_help
+                    exit 1
+                fi
+                ;;
+            -t|--temp-dir)
+                if [ -n "$2" ]; then
+                    readonly TEMP_DIR="$2"
+                    shift 2
+                else
+                    echo_red "Error: Argument required for $1"
+                    show_help
+                    exit 1
+                fi
+                ;;
+            --noclean)
+                CLEAN=false
+                shift 1
+                ;;
+            --force)
+                FORCE=true
+                shift 1
+                ;;
+            *)
+                echo_red "Error: Unknown option: $1"
+                show_help
+                exit 1
+                ;;
+        esac
+    done
+    if [ -z "${MPI_PATH}" ]; then
+        echo_yellow "MPI path was not specified and it is required"
+        show_help
+        exit 0
+    fi
+
+    readonly CLEAN=${CLEAN:=true}
+    readonly FORCE=${FORCE:=false}
+}
+
+
+# ****Warning: patchelf v0.18.0 (currently shipped with EESSI) does not work.****
+# We get v0.17.2
+download_patchelf() {
+    # Temporary directory to save patchelf
+    local tmpdir=$1
+
+    local patchelf_version="0.17.2"
+    local url
+    local curl_opts="-L --silent --show-error --fail"
+
+    url="https://github.com/NixOS/patchelf/releases/download/${patchelf_version}/"
+    url+="patchelf-${patchelf_version}-${EESSI_CPU_FAMILY}.tar.gz"
+
+    local patchelf_path=${tmpdir}/patchelf
+    mkdir ${patchelf_path}
+
+    curl ${url} ${curl_opts} -o ${patchelf_path}/patchelf.tar.gz
+    tar -xf ${patchelf_path}/patchelf.tar.gz -C ${patchelf_path}
+    PATCHELF_BIN=${patchelf_path}/bin/patchelf
+}
+
+# Declaring this var here to use later on the main function
+# This only applies to MPI=>5
+libcuda_exists=0
+
+inject_mpi() {
+    # Temporary directory for injection
+    local tmpdir=$1
+
+    local eessi_ldd="${EESSI_EPREFIX}/usr/bin/ldd"
+    local system_ldd="/usr/bin/ldd"
+
+    local host_injection_mpi_path
+
+    host_injection_mpi_path=${EESSI_SOFTWARE_PATH/versions/host_injections}
+    host_injection_mpi_path+="/rpath_overrides/OpenMPI/system/lib"
+
+    if [ -d ${host_injection_mpi_path} ]; then
+        if [ -n "$(ls -A ${host_injection_mpi_path})" ]; then
+            echo "MPI was already injected"
+            if ${FORCE}; then
+                echo "Forcing new MPI injection"
+            else
+                return 0
+            fi
+        fi
+    fi
+
+    mkdir -p ${host_injection_mpi_path}
+
+    local temp_inject_path="${tmpdir}/mpi_inject"
+    mkdir ${temp_inject_path}
+
+    # Get all library files from openmpi dir
+    find ${MPI_PATH} -type f -name "*.so*" -exec cp {} ${temp_inject_path} \;
```

Comment:
Hmm, this is probably not going to play nice with an OS installed version of OpenMPI as the libraries may well be in the same place as all the other libraries on the system. We should only search for the libraries that make up the compatibility libraries (see https://docs.open-mpi.org/en/v5.0.x/version-numbering.html#shared-library-version-number): 
```
libmpi
libmpi_mpifh
libmpi_usempi_tkr
libmpi_usempi_ignore_tkr
libmpi_usempif08
libmpi_cxx
libmpi_java
liboshmem
```

Reply:
I also wonder what will happen when it tries to load a provider...